test/scripts/parallel-local/test-runner.js

```diff
@@ -0,0 +1,101 @@
+const { exec } = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+let isFirstTest = true;
+
+/**
+ * Ensures the directory exists before executing tests.
+ * @param {string} testType The type of the test.
+ * @returns {void} Does not return a value.
+ */
+const ensureDirectoryExists = (testType) => {
+    const dirPath = path.join('./database/storage', testType);
+    if (!fs.existsSync(dirPath)) {
+        fs.mkdirSync(dirPath, { recursive: true });
+    }
+};
+
+/**
+ * Handles incoming messages to determine test execution workflow.
+ * @param {string|Object} message - Message received from the main process.
+ * @returns {void} Does not return a value.
+ */
+const processMessage = (message) => {
+    if (message === 'no_more_tests') {
+        process.exit();
+    } else {
```

Comment:
I think that you should be explicit on messages, no else. A switch might make sense?
Also, if you have a way to use constants for messages it would be good, and it would be perfect if you find a way to share this constant (for example by importing it into main)

Reply:
Done, sharedConstants.js now handles all shared constants